src/commonMain/kotlin/fr/acinq/lightning/serialization/v1/ChannelState.kt

```diff
@@ -523,20 +522,36 @@ data class WaitForFundingCreated(
 }
 
 @Serializable
-data class InitFunder(
-    @Serializable(with = ByteVector32KSerializer::class) val temporaryChannelId: ByteVector32,
+data class FundingInput(
+    @Serializable(with = TransactionKSerializer::class) val previousTx: Transaction,
+    val outputIndex: Int,
+    @Serializable(with = PrivateKeyKSerializer::class) val privateKey: PrivateKey
+) {
+    constructor(from: fr.acinq.lightning.channel.FundingInput) : this(from.previousTx, from.outputIndex, from.privateKey)
+}
+
+@Serializable
+data class FundingInputs(
     @Serializable(with = SatoshiKSerializer::class) val fundingAmount: Satoshi,
+    val inputs: List<FundingInput>,
+    @Serializable(with = PublicKeyKSerializer::class) val changePubKey: PublicKey?,
+) {
+    constructor(from: fr.acinq.lightning.channel.FundingInputs) : this(from.fundingAmount, from.inputs.map { FundingInput(it) }, from.changePubKey)
+}
+
+@Serializable
+data class InitInitiator(
+    val fundingInputs: FundingInputs,
```

Comment:
IIUC, we can safely change those classes without impacting serialization because this is only used *before* we actually have commitments, so it's never stored in the DB and only used to provide JSON serialization for the app.

Reply:
Yes, the current code does not distinguish between classes that are serializable because we need to display them as JSON for example and classes that are serializable because they need to be persisted in our database.